src/gui/sdlmain.cpp

```diff
@@ -1467,28 +1449,11 @@ void GFX_EndUpdate( const Bit16u *changedLines ) {
 	bool actually_updating = sdl.updating;
 	sdl.updating=false;
 	switch (sdl.desktop.type) {
-	case SCREEN_SURFACE:
-		if (changedLines) {
-			Bitu y = 0, index = 0, rectCount = 0;
-			while (y < sdl.draw.height) {
-				if (!(index & 1)) {
-					y += changedLines[index];
-				} else {
-					SDL_Rect *rect = &sdl.updateRects[rectCount++];
-					rect->x = sdl.clip.x;
-					rect->y = sdl.clip.y + y;
-					rect->w = (Bit16u)sdl.draw.width;
-					rect->h = changedLines[index];
-					y += changedLines[index];
-				}
-				index++;
-			}
-			if (rectCount)
-				SDL_UpdateWindowSurfaceRects(sdl.window, sdl.updateRects, rectCount);
-		}
-		break;
 	case SCREEN_TEXTURE:
-		SDL_UnlockTexture(sdl.texture.texture);
+		SDL_UpdateTexture(sdl.texture.texture,
```

Comment:
`assert(sdl.texture.input_surface)` before 1453? (I suppose these are only needed if the static analyzer flags these.. I'm just been unnecessarily defensive)